gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -358,9 +358,13 @@ public WritableByteChannel create(final StorageResourceId resourceId, CreateObje
      */
 
     Optional<Long> overwriteGeneration =
-        resourceId.hasGenerationId()
-            ? Optional.of(resourceId.getGenerationId())
-            : getWriteGeneration(resourceId, options.overwriteExisting());
+        options.overwriteExisting()
+            ? Optional.absent()
+            : Optional.of(getWriteGeneration(resourceId));
```

Comment:
Aren't setting generation to non-zero here will overwrite object if `options.overwriteExisting()` is `false`?

Reply:
Updated